RoaringBitmap/src/main/java/org/roaringbitmap/longlong/HighLowContainer.java

```diff
@@ -0,0 +1,166 @@
+package org.roaringbitmap.longlong;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import org.roaringbitmap.Container;
+import org.roaringbitmap.art.Art;
+import org.roaringbitmap.art.ContainerIterator;
+import org.roaringbitmap.art.Containers;
+import org.roaringbitmap.art.KeyIterator;
+import org.roaringbitmap.art.LeafNodeIterator;
+
+public class HighLowContainer {
+
+  private Art art;
+  private Containers containers;
+  private static final byte EMPTY_TAG = 0;
+  private static final byte NOT_EMPTY_TAG = 1;
+
+  public HighLowContainer() {
+    art = new Art();
+    containers = new Containers();
+  }
+
+  public Container getContainer(long containerIdx) {
+    return containers.getContainer(containerIdx);
+  }
+
+  public ContainerWithIndex searchContainer(byte[] highPart) {
```

Comment:
Can we use a `long` and an int `shift` instead? Reduce allocation.

Reply:
the high part byte array was fetched from the ART, no additional allocation